From 32616ff188e11913181cf727b26fb0c6357b10f8 Mon Sep 17 00:00:00 2001
From: Johnny Mnemonic <jm@machine-hall.org>
Date: Thu, 5 Jun 2025 15:54:08 +0200
Subject: [PATCH 71/94] ia64: replay "mm: introduce a common definition of
 mk_pte()"

See cb5b13cd6c9237fe5ac978b22453eb3fa098a8d6.

Otherwise we get a multitude of:
```
In file included from ./include/linux/pgtable.h:6,
                 from ./arch/ia64/include/asm/uaccess.h:40,
                 from ./include/linux/uaccess.h:12,
                 from ./include/linux/sched/task.h:13,
                 from ./include/linux/sched/signal.h:9,
                 from ./include/linux/rcuwait.h:6,
                 from ./include/linux/percpu-rwsem.h:7,
                 from ./include/linux/fs.h:34,
                 from ./include/linux/proc_fs.h:10,
                 from init/main.c:18:
./arch/ia64/include/asm/pgtable.h:215:2: error: expected identifier or '(' before '{' token
  215 | ({ pte_t __pte; pte_val(__pte) = ((pfn) << PAGE_SHIFT) | pgprot_val(pgprot); __pte; })
      |  ^
./arch/ia64/include/asm/pgtable.h:220:33: note: in expansion of macro 'pfn_pte'
  220 | #define mk_pte(page, pgprot)    pfn_pte(page_to_pfn(page), (pgprot))
      |                                 ^~~~~~~
./include/linux/mm.h:1786:21: note: in expansion of macro 'mk_pte'
 1786 | static inline pte_t mk_pte(struct page *page, pgprot_t pgprot)
      |                     ^~~~~~
  AS      arch/ia64/kernel/entry.o
```

...during build.
---
 arch/ia64/include/asm/pgtable.h | 2 --
 1 file changed, 2 deletions(-)

diff --git a/arch/ia64/include/asm/pgtable.h b/arch/ia64/include/asm/pgtable.h
index 9be2d2ba6016..7542a4badf06 100644
--- a/arch/ia64/include/asm/pgtable.h
+++ b/arch/ia64/include/asm/pgtable.h
@@ -217,8 +217,6 @@ ia64_phys_addr_valid (unsigned long addr)
 /* Extract pfn from pte.  */
 #define pte_pfn(_pte)		((pte_val(_pte) & _PFN_MASK) >> PAGE_SHIFT)
 
-#define mk_pte(page, pgprot)	pfn_pte(page_to_pfn(page), (pgprot))
-
 /* This takes a physical page address that is used by the remapping functions */
 #define mk_pte_phys(physpage, pgprot) \
 ({ pte_t __pte; pte_val(__pte) = physpage + pgprot_val(pgprot); __pte; })
-- 
2.25.1

